### ISSUE TITLE
games.log broken with plenty of EOF characters

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#926]
Hi, current `games.log` writing is broken.  In many cases, `\0` is used instead of `\n`, and sometime `\0` is added to `\n`. So, to fix the log, sometime `\0` must be substituted by `\n`, and sometime `\0` must just be dropped.

log readers can pipe that to fix the entry:

```sh
tr '\0' '\n' | grep -v '^$'
```

like in this UUOC:

```sh
cat .unvanquished/game/games.log | tr '\0' '\n' | grep -v '^$' | less

```

**Comment by Kangz** [comment]
Do you notice a pattern of where the \0 is output so I can try to reproduce?

**Comment by mbasaglia** [comment]
I think someone mentioned on IRC a similar issue with admin.dat.

**Comment by illwieckz** [comment]
@Kangz, you just have to start a server to reproduce (mere results if you join or start a bot). Almost all lines contain a broken `\0` character. And the `ClientConnect` or `ClientBegin`are the worst ones (no `\n` at all).

**Comment by DolceTriade** [comment]
Might be my \n cleanup that caused this...

**Comment by cu-kai** [comment]
@mbasaglia it was I who was having issues with the admin.dat, this log issue is breaking the New EDGE stats system too.

Here is a screenshot of a snippit of our games.log when opened in Sublime Text 2.

![screen shot 2016-04-05 at 23 01 35](https://cloud.githubusercontent.com/assets/13281185/14299478/94ffb80c-fb82-11e5-9713-a44863b4fdff.png)

**Comment by Kangz** [comment]
@illwieckz, thanks

**Comment by illwieckz** [comment]
seems fixed in 0.50.0
